record_batch.go

```diff
@@ -203,6 +204,9 @@ func (b *RecordBatch) encodeRecords(pe packetEncoder) error {
 	b.recordsLen = len(raw)
 
 	b.compressedRecords, err = compress(b.Codec, b.CompressionLevel, raw)
+	if err != nil {
```

Comment:
🤔

Reply:
again, leftovers of debugging, not sure why go compiler allowed this 🤷‍♂

Reply:
🙏Linter will catch these on CI once I'm done with https://github.com/Shopify/sarama/pull/1573.